backend/app/services/stock_service.py

```
@@ -29,8 +30,49 @@ async def search_stocks(self, query: str, limit: int = 20) -> List[Dict[str, Any
+                    cassandra_service.log_stock_search(
+                        self.context.client_account_id,
+                        self.context.engagement_id,
+                        str(self.context.user_id or "system"),
+                        query,
+                        search_type=search_type,
+                        results_count=len(stocks),
+                        execution_time_ms=execution_time_ms,
+                        source=source,
+                        results=stocks,
+                    )
+                )
```

Ruff (0.14.10)

Store a reference to the return value of `asyncio.create_task`

(RUF006)